lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1537,24 +1538,52 @@ namespace Js
             Assert(oldTypeToPromotedTypeMap && (Js::Var)oldTypeToPromotedTypeMap != scriptContext->GetLibrary()->GetUndefined());
             oldTypeToPromotedTypeMap = reinterpret_cast<TypeTransitionMap*>(oldTypeToPromotedTypeMap);
 
-            if (oldTypeToPromotedTypeMap->TryGetValue(reinterpret_cast<uintptr_t>(oldType), &cachedDynamicType))
+            RecyclerWeakReference<DynamicType>* weakRefCachedtype;
+            
+            if (oldTypeToPromotedTypeMap->TryGetValue(taggedOldType, &weakRefCachedtype))
             {
+                cachedDynamicType = weakRefCachedtype->Get();
```

Comment:
No. When `cachedDynamictype` gets reclaimed, the `strongRef` field of weakReferenceHandle will be set to `nullptr` and will not be reset with reallocation. `Get()` will return `nullptr` if that is the case.
However I made a repro where other way round could happen. `oldType` is cached in dictionary (with tag) and it gets collected and re-allocated with new type. We come down this path and see that `oldType` is present in the map and extracts the `cachedDynamicType` which is wrong.